homeassistant/components/ambient_network/coordinator.py

```diff
@@ -0,0 +1,81 @@
+"""DataUpdateCoordinator for the Ambient Weather Network integration."""
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+from typing import Any, cast
+
+from aioambient import OpenAPI
+from aioambient.errors import RequestError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MAC
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import API_LAST_DATA, DOMAIN, LOGGER, SCAN_INTERVAL
+
+
+class AmbientNetworkDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """The Ambient Network Data Update Coordinator."""
+
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant, api: OpenAPI) -> None:
+        """Initialize the coordinator."""
+        super().__init__(hass, LOGGER, name=DOMAIN, update_interval=SCAN_INTERVAL)
+        self.api = api
+        self.data = {}
+        self.outstanding_errors: list[str] = []
+
+    def _report_error(self, message: str) -> None:
+        """Report an error by logging it. Only log the first occurrence of each error."""
+
+        if message not in self.outstanding_errors:
+            LOGGER.warning(message)
+            self.outstanding_errors.append(message)
+
+    def _clear_errors(self) -> None:
+        """Clear all outstanding errors."""
+
+        if self.outstanding_errors:
+            LOGGER.info(f"Station '{self.config_entry.title}' is back online")
+            self.outstanding_errors = []
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        """Fetch the latest data from the Ambient Network."""
+
+        try:
+            response = await self.api.get_device_details(
+                self.config_entry.data[CONF_MAC]
+            )
+        except RequestError:
```

Comment:
This should raise `UpdateFailed` and all other "failures" below should also use `UpdateFailed` with a proper text to allow the user to see the issues in their logs.

Reply:
Fixed.